From aca4b3e46e2f93e443849a6811fa6c15590b8de3 Mon Sep 17 00:00:00 2001
From: Richard Danter <radanter@gmail.com>
Date: Fri, 9 Jun 2023 14:12:00 +0100
Subject: [PATCH] Revert "ARM: 9290/1: uaccess: Fix KASAN false-positives"

This reverts commit 788f4a3d398757d94e71f2780a2de1cb78f92219.

The upstream patch breaks the RPi BSP builds.

---
 arch/arm/lib/uaccess_with_memcpy.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/lib/uaccess_with_memcpy.c b/arch/arm/lib/uaccess_with_memcpy.c
index c4c5b2c2defd..b483e5713039 100644
--- a/arch/arm/lib/uaccess_with_memcpy.c
+++ b/arch/arm/lib/uaccess_with_memcpy.c
@@ -168,7 +168,7 @@ __copy_to_user_memcpy(void __user *to, const void *from, unsigned long n)
 			tocopy = n;
 
 		ua_flags = uaccess_save_and_enable();
-		__memcpy((void *)to, from, tocopy);
+		memcpy((void *)to, from, tocopy);
 		uaccess_restore(ua_flags);
 		to += tocopy;
 		from += tocopy;
@@ -312,7 +312,7 @@ __clear_user_memset(void __user *addr, unsigned long n)
 			tocopy = n;
 
 		ua_flags = uaccess_save_and_enable();
-		__memset((void *)addr, 0, tocopy);
+		memset((void *)addr, 0, tocopy);
 		uaccess_restore(ua_flags);
 		addr += tocopy;
 		n -= tocopy;
-- 
2.34.1

